README.md

```diff
@@ -1,18 +1,93 @@
-# mock_navigator
+# 🕊 mockingjay
 
-An experimental package that attempts to make it easy to mock Flutter's navigator routes.
+[![Very Good Ventures](https://raw.githubusercontent.com/VeryGoodOpenSource/mockingjay/main/assets/vgv_logo.png)](https://verygood.ventures)
+
+Developed with 💙 by [Very Good Ventures](https://verygood.ventures) 🦄
+
+[![ci](https://github.com/jeroen-meijer/mockingjay/workflows/ci/badge.svg)](https://github.com/jeroen-meijer/mockingjay/actions)
+[![pub package](https://img.shields.io/pub/v/mockingjay.svg)](https://pub.dartlang.org/packages/mockingjay)
+[![License: MIT](https://img.shields.io/badge/license-MIT-blue.svg)](https://opensource.org/licenses/MIT)
+[![style: very good analysis][badge]][badge_link]
+
+---
+
+This is an experimental package that makes it easy to mock, test and verify navigation calls in Flutter. It works in tandem with [`mocktail`][mocktail] and [`mockito`][mockito], allowing you to mock a navigator the same way you would any other object, making it easier to test navigation behavior independently from the UI it's supposed to render.
 
 ## Usage
 
-TODO
+To use the package in your tests, add it to your dev dependencies in your `pubspec.yaml`:
+
+```yaml
+dev_dependencies:
+  mockingjay: ^0.1.0
+```
+
+Then, in your tests, create a `MockNavigator` class like so:
+
+```dart
+import 'package:mocktail/mocktail.dart';
+// If you're using mockito, import that instead.
+// import 'package:mockito/mockito.dart';
+
+class MockNavigator extends Mock
+    with MockNavigatorDiagnosticsMixin
```

Comment:
Why is this mixin needed?

Reply:
Currently looking for a workaround. Removing this mixin from the `class MockNavigator` declaration produces this error, because the `Mock` class override a regular `toString` method signature, and not a diagnosticable one that `NavigatorState` has (which is the class being implemented).
```
'Mock.toString' ('String Function()') isn't a valid concrete implementation of 'Diagnosticable.toString' ('String Function({DiagnosticLevel minLevel})').
```

Reply:
Resolved. Opted to enforce the usage of `mocktail` until further notice.